components/base/kinematicbase/execution.go

```diff
@@ -416,55 +416,86 @@ func (ptgk *ptgBaseKinematics) courseCorrect(
 			// the ending configuration's arc start value.
 			// The connection point is the point along the already-created plan where course correction will rejoin.
 			connectionPoint := arcSteps[solution.stepIdx]
+
+			// Construct deep copy of the connectionPoint to avoid data race
+			connectionPointDeepCopy := arcStep{
```

Comment:
This was better before as a function.  Can we go back to that?  `func copyArcStep(arcStep) arcStep`.  Then we can write a unit test to ensure this function is correct individually